docs/decisions/0010-limit-modulestore-use-in-lms.rst

```diff
@@ -0,0 +1,273 @@
+
+  The ``update_outline_from_modulestore`` is a short function that calls
+  ``get_outline_from_modulestore`` to create a representation of the data that
+  the ``learning_sequences`` app understands (``CourseOutlineData``), and then
+  pushes that data into ``learning_sequences`` via an API method that
+  ``learning_sequences`` exposes (``replace_course_outline``).
+
+  This function also sets custom attributes so that we can better monitor for
+  performance issues and errors.
+
+  Note: One of the things we write is the *version* of the course. This is going
+  to be important for diagnosing what's going on if these writes ever start
+  failing. We get this information from the ``course_version`` attribute on the
+  root ``CourseBlock``, and convert that to a string for convenient storage
+  (it's a BSON object).
+
+Celery Task
+  ``cms.djangoapps.contentstore.outlines.tasks.update_outline_from_modulestore_task``
+
+  This is a simple celery ``@shared_task`` that wraps the call to
+  ``update_outline_from_modulestore``. It's critical to use celery to do this
+  work asynchronously. Even if your code seems to work quickly enough to run
+  in-process, courses can often use obscure features that can drastically
+  increase the time it takes to get data out, and you will almost certainly not
+  be able to comprehensively test for all those situations.
+
+  *You must be aggressive about alerting on task failures*. Publishes are
+  infrequent enough to make it so that certain content-dependent errors will not
+  trigger most error rate alerts. You have to be extremely sensitive to outright
+  failures in your task because you may be blocking the publish for a course.
+
+Signal Handler
+  ``cms.djangoapps.contentstore.outlines.signals.handlers.listen_for_course_publish``
+
+  This is a centralized location where Studio does its post-publish data pushes,
+  but you can also make a separate handler that listens for the same
+  ``course_published`` signal. It's main task is to do some logging and queue
+  the celery task.
+
+Management Task
+  ``cms.djangoapps.contentstore.management.commands.backfill_course_outlines.py``
+  ``cms.djangoapps.contentstore.management.commands.update_course_outline.py``
+
+  Management commands to backfill a group of course outlines or to update one
+  particular command. A few things to note:
+
+  1. These commands live in the Studio process, because they are invoking code
+     that will query the Modulestore.
+  2. The backfill command launches a new celery task for every individual
+     course. This is both to control memory usage (successive Modulestore access
+     across courses will leak a lot of memory), as well as to make it easier to
+     see which courses are taking longer and/or causing errors.
+  3. In the long term, you will want a way to trigger backfills from the Django
+     admin, so that you don't need to file a support ticket every time.
```

Comment:
Once the app-permissions work is done and there's a admin action for backfilling learning sequences, let's come back and update this conversion guide. Ideally, future devs in this situation won't even need to make a management command.